tests/integration/telemetry/stats/prometheus/tcp/stats_tcp_filter_test.go

```diff
@@ -0,0 +1,198 @@
+	cleanupFilterConfig          = "testdata/cleanup.yaml"
+)
+
+var (
+	ist        istio.Instance
+	bookinfoNs namespace.Instance
+	g          galley.Instance
+	ing        ingress.Instance
+	prom       prometheus.Instance
+)
+
+func TestTcpMetric(t *testing.T) { // nolint:interfacer
+	framework.
+		NewTest(t).
+		RequiresEnvironment(environment.Kube).
+		Run(func(ctx framework.TestContext) {
+			addr := ing.HTTPAddress()
+			url := fmt.Sprintf("http://%s/productpage", addr.String())
+			g.ApplyConfigOrFail(
+				t,
+				bookinfoNs,
+				bookinfo.GetDestinationRuleConfigFileOrFail(t, ctx).LoadWithNamespaceOrFail(t, bookinfoNs.Name()),
+				bookinfo.NetworkingTCPDbRule.LoadWithNamespaceOrFail(t, bookinfoNs.Name()),
+			)
+			defer g.DeleteConfig(
+				bookinfoNs,
+				bookinfo.GetDestinationRuleConfigFileOrFail(t, ctx).LoadWithNamespaceOrFail(t, bookinfoNs.Name()),
+				bookinfo.NetworkingTCPDbRule.LoadWithNamespaceOrFail(t, bookinfoNs.Name()),
+			)
+
+			systemNM := namespace.ClaimSystemNamespaceOrFail(ctx, ctx)
+			// TODO(gargnupur): Use istioctl to install metadata exchange and stats filters.
+			mxc, err := file.AsString(metadataExchangeFilterConfig)
+			if err != nil {
+				t.Errorf("unable to load config %s, err:%v", metadataExchangeFilterConfig, err)
+			}
+			stats, err := file.AsString(statsFilterConfig)
+			if err != nil {
+				t.Errorf("unable to load config %s, err:%v", statsFilterConfig, err)
+			}
+
+			cleanup, err := file.AsString(cleanupFilterConfig)
+			if err != nil {
+				t.Errorf("unable to load config %s, err:%v", metadataExchangeFilterConfig, err)
+			}
+
+			g.ApplyConfigOrFail(
+				t,
+				systemNM,
+				mxc,
+				stats,
+				cleanup,
+			)
+			defer g.DeleteConfig(
+				systemNM,
+				mxc,
+				stats,
+				cleanup,
+			)
+
+			util.AllowRuleSync(t)
+
+			destinationQuery := buildQuery()
+			retry.UntilSuccessOrFail(t, func() error {
+				util.SendTraffic(ing, t, "Sending traffic", url, "", 200)
+				// TODO(gargnupur): Use TCP metrics like in Telemetry V1 (https://github.com/istio/istio/issues/20283)
+				if err := util_prometheus.QueryPrometheus(t, destinationQuery, prom); err != nil {
+					t.Logf("prometheus values for istio_requests_total: \n%s", util.PromDump(prom, "istio_requests_total"))
+					return err
+				}
+				return nil
+			}, retry.Delay(3*time.Second), retry.Timeout(80*time.Second))
+		})
+}
+
+func TestMain(m *testing.M) {
+	framework.
+		NewSuite("stats_tcp_filter", m).
+		RequireEnvironment(environment.Kube).
+		Label(label.CustomSetup).
+		SetupOnEnv(environment.Kube, istio.Setup(&ist, setupConfig)).
+		Setup(testsetup).
+		Run()
+}
+
+func setupConfig(cfg *istio.Config) {
+	if cfg == nil {
+		return
+	}
+	// disable mixer telemetry and enable telemetry v2
+	// This turns on telemetry v2 for both HTTP and TCP.
+	cfg.Values["telemetry.enabled"] = "true"
+	cfg.Values["telemetry.v1.enabled"] = "false"
+	cfg.Values["telemetry.v2.enabled"] = "true"
```

Comment:
this flag is default off and the effect only are showed in this particular test, right? might worth to try a default on (WIP pr, no need for merge) to see if other features like permissive or sniffing still works from CI tests running results

Reply:
Have a PR for turning this default on.. https://github.com/istio/istio/pull/20258 (will work to fix the conflicts from it)

Regarding permissive or sniffing related tests, they should get ALPN on by default because that is dependent on value of PILOT_ENABLE_TCP_METADATA_EXCHANGE which is default on. This test, tests the mixer v2 part of it.